syntax-markdown-commonmark12/src/main/java/org/xwiki/contrib/rendering/markdown/commonmark12/internal/parser/DeepInlineHTMLPostProcessor.java

```diff
@@ -0,0 +1,112 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.contrib.rendering.markdown.commonmark12.internal.parser;
+
+import org.jetbrains.annotations.NotNull;
+
+import com.vladsch.flexmark.ast.HtmlInline;
+import com.vladsch.flexmark.parser.block.NodePostProcessor;
+import com.vladsch.flexmark.parser.block.NodePostProcessorFactory;
+import com.vladsch.flexmark.util.ast.Document;
+import com.vladsch.flexmark.util.ast.Node;
+import com.vladsch.flexmark.util.ast.NodeTracker;
+import com.vladsch.flexmark.util.data.DataHolder;
+
+/**
+ * Detects and merges nodes that are content of inline HTML open and close tags.
+ * In a situation where a paragraph contains an open tag without any corresponding closing tag, the rest of the
+ * paragraph content will be treated as a single HtmlInline node.
+ *
+ * @version $Id$
+ * @since 8.9
+ */
+public class DeepInlineHTMLPostProcessor extends NodePostProcessor
+{
+    private static final String HTML_OPEN_PREFIX = "<";
+    private static final String HTML_CLOSE_PREFIX = "</";
+    private static final String HTML_SELF_SUFFIX = "/>";
+    private static final String HTML_TAG_SUFFIX = "\\s+|>";
+
+    /**
+     * Factory class for DeepInlineHTMLPostProcessor.
+     */
+    public static class Factory extends NodePostProcessorFactory
+    {
+        /**
+         * Factory constructor.
+         *
+         * @param options parser configuration
+         */
+        public Factory(DataHolder options)
+        {
+            super(false);
+            addNodes(HtmlInline.class);
+        }
+
+        @NotNull
+        @Override
+        public NodePostProcessor apply(@NotNull Document document)
+        {
+            return new DeepInlineHTMLPostProcessor();
+        }
+    }
+
+    @Override
+    public void process(@NotNull NodeTracker nodeTracker, @NotNull Node node)
+    {
+        // We store the number of nested open tags found during the processing and match it to the number of closing
+        // tags.
+        int toFind = 1;
+        int nFound = 0;
+
+        Node nextNode = node.getNext();
+        String inlineStartText = node.getChars().toString();
+
+        // We return early if we found a single closing tag, or a self-closing one.
+        if (inlineStartText.startsWith(HTML_CLOSE_PREFIX) || inlineStartText.endsWith(HTML_SELF_SUFFIX)) {
+            return;
+        }
+
+        // We keep the name of the tag until the first whitespace character and remove the angle brackets.
+        String inlineStartTag = inlineStartText.split(HTML_TAG_SUFFIX)[0].substring(1);
```

Comment:
It would be best here to use either a "static final" pattern, see [this Sonar rule](https://rules.sonarsource.com/java/RSPEC-4248/), or use [`StringUtils.split`](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#split-java.lang.String-java.lang.String-) which doesn't use regular expressions and also supports several separator characters.